transport/http/outbound.go

```diff
@@ -41,61 +45,61 @@ var (
 	errOutboundNotStarted     = errors.ErrOutboundNotStarted("http.Outbound")
 )
 
-type outboundConfig struct {
-	keepAlive time.Duration
-}
+// NewOutbound is deprecated
+// TODO rename this func NewOutboundFromURL
+func NewOutbound(urlStr string) transport.Outbound {
+	agent := NewDefaultAgent()
 
-var defaultConfig = outboundConfig{keepAlive: 30 * time.Second}
+	scheme, hp, path := parseURL(urlStr)
 
-// OutboundOption customizes the behavior of an HTTP outbound.
-type OutboundOption func(*outboundConfig)
+	peerID := hostport.NewPeerIdentifier(hp)
+	peerList := peerlist.NewSingle(peerID, agent)
 
-// KeepAlive specifies the keep-alive period for the network connection. If
-// zero, keep-alives are disabled.
-//
-// Defaults to 30 seconds.
-func KeepAlive(t time.Duration) OutboundOption {
-	return func(c *outboundConfig) {
-		c.keepAlive = t
-	}
+	return NewPeerListOutbound(peerList, path, scheme)
 }
 
-// NewOutbound builds a new HTTP outbound that sends requests to the given
-// URL.
-func NewOutbound(url string, opts ...OutboundOption) transport.Outbound {
-	cfg := defaultConfig
-	for _, o := range opts {
-		o(&cfg)
+func parseURL(urlStr string) (string, string, string) {
```

Comment:
When returning multiple things of the same type like this, use named
parameters to document what you're returning:

    parseURL(urlStr string) (scheme, host, path string) {
        // ..
    }